### PR TITLE
[INF] Set independent environment for building documentation

### PR DESCRIPTION
**Pull request by Zeroto521**
# PR Description

Please describe the changes proposed in the pull request:

/environment-dev.yml is complex.
It's used for building doc, setting developer local env, and setting ci test env.
This file becomes big and big and hard to understand.

Part of #1133.

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: .github/workflows/docs-preview.yml [hunk 1]
```diff
@@ -22,16 +22,12 @@ jobs:
         with:
           auto-update-conda: true
           miniforge-variant: Mambaforge
-          channels: conda-forge
-          activate-environment: pyjanitor-dev
```
Comment (Zeroto521) [inline .github/workflows/docs-preview.yml:26]: The value of channels and activate-environment use the variable of mkdocs/environment.yml

File: .github/workflows/docs-preview.yml [hunk 2]
```diff
@@ -22,16 +22,12 @@ jobs:
...
-          environment-file: environment-dev.yml
+          environment-file: mkdocs/environment.yml
           use-mamba: true
 
       - name: Build docs
         run: |
-          conda activate pyjanitor-dev
```
Comment (Zeroto521) [inline .github/workflows/docs-preview.yml:32]: default will activate that env

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,7 @@
 -   [ENH] `dropna` parameter added to `pivot_longer`. Issue #1132 @samukweku
 -   [INF] Update `mkdocstrings` version and to fit its new coming features. PR #1138 @Zeroto521
 -   [BUG] Force `math.softmax` returning `Series`. PR #1139 @Zeroto521
+-   [INF] Set indenpend environment for building documentation. PR #1141 @Zeroto521
```
Comment (ericmjl) [inline CHANGELOG.md]:
```suggestion
-   [INF] Set independent environment for building documentation. PR #1141 @Zeroto521
```

**Review by ericmjl — CHANGES_REQUESTED** [review]
Only one change request, and then let's merge!

**Comment by Zeroto521** [comment]
Why this test case (tests/functions/test_conditional_join.py::test_multiple_eqs_extension_array) failed in https://github.com/pyjanitor-devs/pyjanitor/runs/7761821318?check_suite_focus=true but passed now?

**Comment by ericmjl** [comment]
> Why this test case (tests/functions/test_conditional_join.py::test_multiple_eqs_extension_array) failed in https://github.com/pyjanitor-devs/pyjanitor/runs/7761821318?check_suite_focus=true but passed now?

I'm not sure, @Zeroto521 - some tests may be flaky, I guess. Usually a re-run of the CI might do the trick.

With everything green, I'm happy to merge!

**Review by ericmjl — APPROVED** [review]

**Pull request merged**